### PR TITLE
SBOM support in "cosign attach" and "cosign download sbom".

### PR DESCRIPTION
**Pull request by dlorenc**
Signed-off-by: Dan Lorenc <dlorenc@google.com>

### CODE REVIEW

File: cmd/cosign/cli/attach/sbom.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	if err != nil {
+		return err
+	}
+
+	m, err := img.Manifest()
+	if err != nil {
+		return err
+	}
+	// TODO: Fix me.
+	m.Config.MediaType = "application/vnd.sbom.config.v1+json"
```
Comment (dlorenc) [inline cmd/cosign/cli/attach/sbom.go:85]: @jonjohnsonjr what do i put here

Comment (jonjohnsonjr) [reply cmd/cosign/cli/attach/sbom.go]: Is there any useful configuration you'd want to put in there?

Comment (dlorenc) [reply cmd/cosign/cli/attach/sbom.go]: No. I could eventually see the need for multiple SBOMs (one for the overall image, then some for sub-parts), but that would get tackled via annotations on each layer to indicate which portion of the image the SBOM is intended to cover.

File: cmd/cosign/cli/attach/sbom.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	})
+	if err != nil {
+		return err
+	}
+
+	m, err := img.Manifest()
+	if err != nil {
+		return err
+	}
+	m.Config.MediaType = "text/spdx"
```
Comment (samj1912) [inline cmd/cosign/cli/attach/sbom.go:83]: Is it possible to support both cyclonedx and spdx here?

Comment (dlorenc) [reply cmd/cosign/cli/attach/sbom.go]: Yup! added

**Comment by dlorenc** [comment]
Dropping the WIP, this is ready to go!

**Review by priyawadhwa — APPROVED** [review]

**Pull request merged**